docs/controller.md

```diff
@@ -1,3 +1,85 @@
 # Controller UI
 
-WIP.
+This provides:
+
+- An index page containing:
+    - a table displaying the current state of the finite state machine (FSM) and buttons
+    to run the possible transitions.
+    - a collapsible tree showing the running applications and their relationship.
+    - a feed showing broadcast messages from the topic `erskafka-reporting`.
+- A view displaying a more detailed table with the application tree
+
+The view functions for this UI are split into two categories:
+
+- `pages` that load a full page.
+- `partials` that load items within a page.
+
+## Index Page
+
+The index page contains three columns, the contents for each of them contained in a
+separate template file and loaded via a partial view. Left to right, the columns contain
+the final state machine, the applications tree and the message feed.
+
+### Finite State Machine (FSM)
+
+The finite state machine describes the current state of the system, the possible transitions
+from this state and where they lead, with all other possible transitions and states greyed out.
+It represents the states and the main transitions described in the [drunc documentation], excluding the sequences.
+
+The FSM diagram is built with a `django-tables2`, with the elements for each row and columns
+defined in the backend based on a **hardcoded** version of the real FSM used by `drunc`.
+This is done this way since `drunc` provides no way of extracting the architecture of the
+FSM machine dynamically, which also means that, potentially, they both (frontend and drunc)
+might become incompatible if the actual FSM is changed but the hardcoded version in the
+frontend is not updated accordingly.
+
+When clicking in a transition, a pop-up modal dialog opens to input the arguments (required
+or optional) to run the transition, and to confirm it. The code for the dialog is contained
+in another partial view and template. The arguments required for each transition are pulled dynamically from
+`drunc` and put together in a `django form`.
```

Comment:
Did you mean `django form` to be a link?

Reply:
Nope. I just put the closing tick in the wrong place. It was meant to cover just `django`.